### PR TITLE
Support Put Roles API customization

### PR DESCRIPTION
**Pull request by n1v0lg**
This PR adds support for injecting a `PutRoleRequest` builder to allow us to customize Put Roles API behavior. It also includes a few related changes:

1. Renames `responseRestricted` methods and vars to `pathRestricted` in the operator route check. This is more appropriate since the request or response may be restricted. To avoid breaking serverless CI, I'll rename the actual value in a follow up (where that rename will be the only change and thus easier to rule out other causes for test failures).
2. Exposes a few convenient methods 
3. Adds an `exists()` check for role names on the `FileRolesStore`

Relates: ES-7825.

### CODE REVIEW

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Comment by n1v0lg** [comment]
@elasticmachine update branch

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/role/PutRoleRequestBuilderFactory.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import java.util.function.Predicate;
+
+public interface PutRoleRequestBuilderFactory {
+    PutRoleRequestBuilder create(Client client, boolean restrictRequest, Predicate<String> fileRolesStoreNameChecker);
+
+    class Default implements PutRoleRequestBuilderFactory {
+        @Override
+        public PutRoleRequestBuilder create(Client client, boolean restrictRequest, Predicate<String> fileRolesStoreNameChecker) {
+            return new PutRoleRequestBuilder(client);
```
Comment (slobodanadamovic) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/role/PutRoleRequestBuilderFactory.java:20]: nit: would be helpful to add a brief javadoc on why the default implementation ignores restrictions

**Review by slobodanadamovic — APPROVED** [review]
LGTM

The IoC change turned out pretty neatly!

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Pull request merged**